pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -256,3 +259,44 @@ func (r *Reconciler) ValidateHardware(ctx context.Context, log logr.Logger, clus
 
 	return controller.Result{}, nil
 }
+
+// CheckRufioMachinesContactable checks to ensure all the Rufio machines condition contactable is True.
+func (r *Reconciler) CheckRufioMachinesContactable(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
```

Comment:
What's the rationale for deviating from `Validate`?

Reply:
Um I dont have a strong preference here. My rational was we have `CheckControlPlaneReady` which checks on a conditions and hence since we do the same here I named it `Check`.

I can make it validate 👍